master/internal/api_runs.go

```diff
@@ -779,3 +779,137 @@ func archiveUnarchiveAction(ctx context.Context, archive bool, runIDs []int32,
 
 	return results, nil
 }
+
+func (a *apiServer) PauseRuns(ctx context.Context, req *apiv1.PauseRunsRequest,
+) (*apiv1.PauseRunsResponse, error) {
+	results, err := pauseResumeAction(ctx, true, req.ProjectId, req.RunIds, req.Filter)
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.PauseRunsResponse{Results: results}, nil
+}
+
+func (a *apiServer) ResumeRuns(ctx context.Context, req *apiv1.ResumeRunsRequest,
+) (*apiv1.ResumeRunsResponse, error) {
+	results, err := pauseResumeAction(ctx, false, req.ProjectId, req.RunIds, req.Filter)
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.ResumeRunsResponse{Results: results}, nil
+}
+
+func pauseResumeAction(ctx context.Context, isPause bool, projectID int32,
+	runIds []int32, filter *string) (
+	[]*apiv1.RunActionResult, error,
+) {
+	if len(runIds) > 0 && filter != nil {
+		return nil, fmt.Errorf("if filter is provided run id list must be empty")
+	}
+	// Get experiment ids
+	var err error
+	var runCandidates []runCandidateResult
+	getQ := db.Bun().NewSelect().
+		ModelTableExpr("runs AS r").
+		Model(&runCandidates).
+		Column("r.id").
+		ColumnExpr("COALESCE((r.archived OR e.archived OR p.archived OR w.archived), FALSE) AS archived").
+		ColumnExpr("r.experiment_id as exp_id").
+		ColumnExpr("((SELECT COUNT(*) FROM runs r WHERE e.id = r.experiment_id) > 1) as is_multitrial").
```

Comment:
this needs to look at the searcher type not at the number of runs. asha doesn't create all its trials at once.

Reply:
Updated

Reply:
we should probably make sure `searchRuns` also behaves like this so the value is consistent throughout. should i make a ticket?

Reply:
Created ticket here: https://hpe-aiatscale.atlassian.net/browse/ET-296